src/grt/src/fastroute/src/utility.cpp

```diff
@@ -96,6 +96,13 @@ void FastRouteCore::ConvertToFull3DType2()
 
 static bool compareNetPins(const OrderNetPin& a, const OrderNetPin& b)
 {
+  // First priority: NDR nets (lower is better, so NDR nets with priority 0 come
+  // first)
+  if (a.ndr_priority != b.ndr_priority) {
+    return a.ndr_priority < b.ndr_priority;
+  }
```

Comment:
You can add this ndr_priority to the std::tie below.

It would look like this:
```
return std::tie(a.ndr_priority, a.length_per_pin, a.minX, a.treeIndex)
         < std::tie(b.ndr_priority, b.length_per_pin, b.minX, b.treeIndex);
```